lib/subliminal/providers/legendastv.py

```diff
@@ -416,6 +431,9 @@ def query(self, language, title, season=None, episode=None, year=None):
                     # cache the releases
                     region.set(releases_key.format(archive_id=a.id), releases)
 
+                    # cache the subtitle archive name
+                    region.set(archive_name_key.format(archive_id=a.id), a.name)
```

Comment:
We shouldn't have two different caches: one for release names and another for archive name. 
Probably the only thing you need to change is the cache key to use archive id and name